datalad/metadata/search.py

```diff
@@ -1304,6 +1314,14 @@ class Search(Interface):
             doc="""if given, the formal query that was generated from the given
             query string is shown, but not actually executed. This is mostly useful
             for debugging purposes."""),
+        use_metadata=Parameter(
```

Comment:
let me think about it more but ATM I dislike `use_metadata` which immediately suggested me being `bool` and then it makes little sense since search works only on metadata. Needed me to get to definition of `use_metadata` here (from points of use above) to understand what it actually means.
Why such confusion: Even though we do have `@use_casssette(cassette_name)`, typically anything `use_` is `bool`.  So may be something like `metadata_type` or `metadata_format` would be a better name and align better with its docstring?

Reply:
Changed the parameter name from `use_metadata` to `metadata_source`.